### PR TITLE
Backport guzzle fixes

### PR DESCRIPTION
**Pull request by CarlSchwan**
See https://github.com/nextcloud/3rdparty/pull/1129

### CODE REVIEW

**Review by miaulalala — APPROVED** [review]

**Review by AndyScherzinger — APPROVED** [review]

**Comment by nickvergessen** [comment]
Needs a rebase to the merge commit once 3rdparty is in

File: apps/dav/tests/unit/Connector/Sabre/RequestTest/RequestTestCase.php [hunk 1]
```diff
@@ -61,6 +62,7 @@ protected function setUp(): void {
 		$this->serverFactory = new ServerFactory(
 			\OC::$server->getConfig(),
 			\OC::$server->getLogger(),
+			\OC::$server->get(LoggerInterface::class),
```
Comment (nickvergessen) [inline apps/dav/tests/unit/Connector/Sabre/RequestTest/RequestTestCase.php:65]: Is this really related?

Comment (CarlSchwan) [reply apps/dav/tests/unit/Connector/Sabre/RequestTest/RequestTestCase.php:65]: Looks like an unrelated local change that I committed by error here :(
